frontend/src/components/Playback/PlayerElement.vue

```diff
@@ -1,66 +1,180 @@
  */
-const teleportTarget = computed<'body' | '.video-container'>(() =>
-  mediaElement.value === 'audio' ? 'body' : '.video-container'
-);
+const teleportTarget = computed<
+  'body' | '.fullscreen-video-container' | '.minimized-video-container'
+>(() => {
+  if (playbackManager.currentlyPlayingMediaElement === 'audio') {
+    return 'body';
+  } else if (
+    playbackManager.currentlyPlayingMediaElement === 'video' &&
+    playerElement.isFullscreenMounted
+  ) {
+    return '.fullscreen-video-container';
+  } else if (
+    playbackManager.currentlyPlayingMediaElement === 'video' &&
+    playerElement.isPiPMounted
+  ) {
+    return '.minimized-video-container';
+  } else {
+    return 'body';
+  }
+});
 
 const posterUrl = computed<string>(() =>
-  !isNil(playbackManager.currentItem) && mediaElement.value === 'video'
+  !isNil(playbackManager.currentItem) &&
+  playbackManager.currentlyPlayingMediaElement === 'video'
     ? getImageInfo(playbackManager.currentItem, {
         preferBackdrop: true
       }).url || ''
     : ''
 );
 
+watch(
+  () => playbackManager.currentSubtitleStreamIndex,
+  () => playerElement.applyCurrentSubtitle()
+);
+
+/**
+ * Called by the media element when the playback is ready
+ */
+function onCanPlay(): void {
+  if (playbackManager.currentlyPlayingMediaType === 'Video') {
+    playerElement.applyCurrentSubtitle();
+  }
+}
+
+/**
+ * Callback for when HLS.js gets an error
+ */
+function onHlsEror(event: Events.ERROR, data: ErrorData): void {
+  if (data.fatal) {
+    switch (data.type) {
+      case ErrorTypes.NETWORK_ERROR: {
+        // try to recover network error
+        console.log('fatal network error encountered, try to recover');
+        hls?.startLoad();
+        break;
+      }
+      case ErrorTypes.MEDIA_ERROR: {
+        console.log('fatal media error encountered, try to recover');
+        hls?.recoverMediaError();
+        break;
+      }
+      default: {
+        // cannot recover
+        hls?.destroy();
+        break;
+      }
+    }
+  }
+}
+
 watch(
   () => playbackManager.currentSourceUrl,
-  () => {
-    if (
-      mediaElementRef.value &&
-      playbackManager.currentSourceUrl &&
-      playbackManager.currentMediaSource?.SupportsDirectPlay &&
-      playbackManager.currentlyPlayingMediaType === 'Audio'
-    ) {
-      mediaElementRef.value.src = playbackManager.currentSourceUrl;
+  async () => {
+    if (playbackManager.currentSourceUrl) {
+      /**
+       * Wait for nextTick to have the DOM updated accordingly
+       */
+      await nextTick();
+
+      try {
+        if (!mediaElementRef.value) {
+          throw new Error('No media element found');
+        }
+
+        if (
+          (playbackManager.currentMediaSource?.SupportsDirectPlay &&
+            playbackManager.currentlyPlayingMediaType === 'Audio') ||
+          (mediaElementRef.value.canPlayType('application/vnd.apple.mpegurl') &&
+            playbackManager.currentlyPlayingMediaType === 'Video')
```

Comment:
With my suggestions above, this check would just be a ``if (isNativeHlsSupported)`` in the else if ``currentlyPlayingMediaType === 'Video'`` block. Much simpler!

Reply:
Done